src/transforms/bin.js

```diff
@@ -4,23 +4,23 @@ import {offset} from "../style.js";
 import {maybeGroup, maybeOutputs, maybeReduce, maybeSubgroup, reduceIdentity} from "./group.js";
 
 // Group on {z, fill, stroke}, then optionally on y, then bin x.
-export function binX(outputs, {inset, insetLeft, insetRight, ...options} = {}) {
+export function binX(outputs = {y: "count"}, {inset, insetLeft, insetRight, ...options} = {}) {
   let {x, y} = options;
   x = maybeBinValue(x, options, identity);
   ([insetLeft, insetRight] = maybeInset(inset, insetLeft, insetRight));
   return binn(x, null, null, y, outputs, {inset, insetLeft, insetRight, ...options});
 }
 
 // Group on {z, fill, stroke}, then optionally on x, then bin y.
-export function binY(outputs, {inset, insetTop, insetBottom, ...options} = {}) {
+export function binY(outputs = {x: "count"}, {inset, insetTop, insetBottom, ...options} = {}) {
   let {x, y} = options;
   y = maybeBinValue(y, options, identity);
   ([insetTop, insetBottom] = maybeInset(inset, insetTop, insetBottom));
   return binn(null, y, x, null, outputs, {inset, insetTop, insetBottom, ...options});
 }
 
 // Group on {z, fill, stroke}, then bin on x and y.
-export function bin(outputs, {inset, insetTop, insetRight, insetBottom, insetLeft, ...options} = {}) {
+export function bin(outputs = {r: "count", fill: "count"}, {inset, insetTop, insetRight, insetBottom, insetLeft, ...options} = {}) {
```

Comment:
```suggestion
export function bin(outputs = {fill: "count"}, {inset, insetTop, insetRight, insetBottom, insetLeft, ...options} = {}) {
```

Are you sure you wouldn’t prefer only providing a *fill* output by default, since it works with both rect and dot and avoids a redundant encoding in the case of dot? I’d prefer `{r: "count"}` for the dot case, but I don’t see how we’d infer how it will be consumed.

Reply:
the simpler the better